src/plugins/pagination.js

```diff
@@ -174,7 +175,7 @@ module.exports = function paginationPlugin (bookshelf) {
         });
         qb.countDistinct.apply(qb, [`${tableName}.${idAttribute}`]);
 
-      }).fetchAll().then(result => {
+      }).fetchAll({transacting}).then(result => {
```

Comment:
I think passing all options might be a problem, as this includes passing things like `require`. I'd argue that the `require` must apply to the `fetchAll` of the rows you want, not the count. I don't know if this matters in practice, or why the count is using a `fetchAll` rather than a `fetch` anyway. It would also be passing `withRelated`, which will fail in the worst case or be wasteful in the best case. Maybe just blacklisting some options to exclude could work, though it presents some forward compatibility issues.